storage/src/test/java/org/apache/kafka/tiered/storage/integration/TransactionsWithTieredStoreTest.java

```diff
@@ -0,0 +1,135 @@
+
+import kafka.api.TransactionsTest;
+import kafka.server.HostedPartition;
+import kafka.server.KafkaBroker;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.test.TestUtils;
+import org.apache.kafka.tiered.storage.utils.BrokerLocalStorage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import scala.collection.JavaConverters;
+import scala.collection.Seq;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import static org.apache.kafka.tiered.storage.utils.TieredStorageTestUtils.createPropsForRemoteStorage;
+import static org.apache.kafka.tiered.storage.utils.TieredStorageTestUtils.createTopicConfigForRemoteStorage;
+import static org.apache.kafka.tiered.storage.utils.TieredStorageTestUtils.STORAGE_WAIT_TIMEOUT_SEC;
+
+public class TransactionsWithTieredStoreTest extends TransactionsTest {
+
+    private String testClassName;
+    private String storageDirPath;
+
+    @BeforeEach
+    @Override
+    public void setUp(TestInfo testInfo) {
+        testClassName = testInfo.getTestClass().get().getSimpleName().toLowerCase(Locale.getDefault());
+        storageDirPath = TestUtils.tempDirectory("kafka-remote-tier-" + testClassName).getAbsolutePath();
+        super.setUp(testInfo);
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public void modifyConfigs(Seq<Properties> props) {
+        for (Properties p : JavaConverters.seqAsJavaList(props)) {
+            p.putAll(overridingProps());
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public Seq<Properties> kraftControllerConfigs() {
+        return JavaConverters.asScalaBuffer(Collections.singletonList(overridingProps())).toSeq();
+    }
+
+    private Properties overridingProps() {
+        int numRemoteLogMetadataPartitions = 3;
+        boolean deleteOnClose = false;
+        return createPropsForRemoteStorage(testClassName, storageDirPath, brokerCount(),
+                numRemoteLogMetadataPartitions, deleteOnClose);
+    }
+
+    @Override
+    public Properties topicConfig() {
+        boolean enableRemoteStorage = true;
+        long localLogRetentionBytes = 1;
+        int maxBatchCountPerSegment = 1;
+        Properties overridingTopicProps = super.topicConfig();
+        overridingTopicProps.putAll(createTopicConfigForRemoteStorage(
+                enableRemoteStorage, localLogRetentionBytes, maxBatchCountPerSegment));
+        return overridingTopicProps;
+    }
+
+    // NOTE: Not able to refer TestInfoUtils#TestWithParameterizedQuorumName() in the ParameterizedTest name.
+    @ParameterizedTest(name = "{displayName}.quorum={0}")
+    @ValueSource(strings = {"zk", "kraft"})
+    public void testFencingOnTransactionExpiration(String quorum) {
```

Comment:
Disabled the `testFencingOnTransactionExpiration` test. Can we take this up in the next PR?

Reply:
Filed https://issues.apache.org/jira/browse/KAFKA-15453 as a followup. If there is already a JIRA, please close this as a duplicate.